plugins/inputs/prometheus/prometheus.go

```diff
@@ -40,27 +47,31 @@ func (p *Prometheus) Gather(_ telegraf.Accumulator) error {
 func (p *Prometheus) Start(accIn telegraf.Accumulator) error {
 	mth := NewMetricsTypeHandler()
 	receiver := &metricsReceiver{pmbCh: p.mbCh}
-	handler := &metricsHandler{mbCh: p.mbCh,
+	handler := &metricsHandler{
+		mbCh:        p.mbCh,
 		acc:         accIn,
 		calculator:  NewCalculator(),
 		filter:      NewMetricsFilter(),
 		clusterName: p.ClusterName,
 		mtHandler:   mth,
 	}
 
-	ecssd := &ecsservicediscovery.ServiceDiscovery{Config: p.ECSSDConfig}
+	var configurer *awsmiddleware.Configurer
+	var ecssd *ecsservicediscovery.ServiceDiscovery
 
-	// Start ECS Service Discovery when in ECS
-	// https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ContainerInsights-Prometheus-Setup-autodiscovery-ecs.html
+	if p.middleware != nil {
+		configurer = awsmiddleware.NewConfigurer(p.middleware.Handlers())
+		if configurer != nil {
+			ecssd = &ecsservicediscovery.ServiceDiscovery{Config: p.ECSSDConfig, Configurer: configurer}
+
+		} else {
+			ecssd = &ecsservicediscovery.ServiceDiscovery{Config: p.ECSSDConfig}
+		}
```

Comment:
nit: Don't need the if. If `configurer` is nil, then it's already equivalent to `&ecsservicediscovery.ServiceDiscovery{Config: p.ECSSDConfig}`